### PR TITLE
[test] test for InformationController and EventController

### PR DESCRIPTION
**Pull request by AntenehAdmasu**
Comments from the previous PR addressed
Unit tests are written for Information and Event Controllers

### CODE REVIEW

File: Backend/controllers/InformationController.js [hunk 1]
```diff
@@ -5,10 +5,12 @@ const _ = require('lodash')
 exports.addInformation = async(req, res) => {
     const information = new Information(_.pick(req.body, ["title", "note"]))
     await information.save()
-    if(req.body.description){
-        await Information.updateOne({_id: information.id}, {$push: { description: req.body.description }});
+    const { description } = req.body
+    if(description){
```
Comment (Khalid-Sultan) [inline Backend/controllers/InformationController.js:9]: Could we have one save operation instead?

File: Backend/tests/EventControllerTests.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            start: "2020-11-17T18:14:26.178Z",
+            end: "2020-11-17T18:15:46.178Z",
+            description: "Openning camp program take 2 for interview preparation",
+            eventType: "CONTEST",
+            gallery: "some html text" 
+        });
+        await event.save();
+    });
+    afterEach(async () => {
+        await Event.deleteMany({});
```
Comment (Khalid-Sultan) [inline Backend/tests/EventControllerTests.js:30]: Instead of deleteMany operation that might accidentally erase every data in the system, perhaps specifying the element itself is safer.

**Review by Khalid-Sultan — COMMENTED** [review]
One or two comments but other than that Excellent code coverage and consideration of edge cases :star:

File: Backend/tests/EventControllerTests.js (hunk 2 above)
Comment (MohammedAlewi) [reply Backend/tests/EventControllerTests.js:30]: I think if we didn't say deleteMany() at line 43 the assertion will fail right? :)

File: Backend/controllers/InformationController.js (hunk 1 above)
Comment (AntenehAdmasu) [reply Backend/controllers/InformationController.js:9]: yea yea sure

File: Backend/tests/EventControllerTests.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            eventType: "CONTEST",
+            gallery: "Some html text" 
+        };
+        it("It should add event ", async () => {   
+            let response = await chai
+                .request(server)
+                .post("/api/event")
+                .send(eventObject);
+            expect(response).to.have.status(201);
+            expect(response.body).to.be.a("object");
```
Comment (Infinidrix) [inline Backend/tests/EventControllerTests.js:90]: Should we delete the event after success so it doesn't clutter the test DB?

File: Backend/tests/EventControllerTests.js (hunk 2 above)
Comment (AntenehAdmasu) [reply Backend/tests/EventControllerTests.js:30]: Yea it might. For post method tests, we have saved a new document. So, incase that operation is interrupted after saving, the saved document might stay there, and as Mame said the assertion on line 43 fails @Khalid-Sultan .

File: Backend/tests/EventControllerTests.js (hunk 3 above)
Comment (Infinidrix) [reply Backend/tests/EventControllerTests.js:90]: Same for Information test

**Review by Infinidrix — APPROVED** [review]
Looks good to me 🥇

File: Backend/tests/EventControllerTests.js (hunk 3 above)
Comment (Infinidrix) [reply Backend/tests/EventControllerTests.js:90]: Never mind saw the above discussion 👍

File: Backend/tests/EventControllerTests.js (hunk 2 above)
Comment (Khalid-Sultan) [reply Backend/tests/EventControllerTests.js:30]: Shouldn't we do cleanups for post methods? Like right after we asserted a post method is working correctly, we can immediately delete the new document using findByIdAndDelete()

**Review by MohammedAlewi — APPROVED** [review]

File: Backend/tests/EventControllerTests.js (hunk 3 above)
Comment (AntenehAdmasu) [reply Backend/tests/EventControllerTests.js:90]: I was thinking to handle that in the `afterEach()` hook. Otherwise, if this doesn't execute completely(reach the delete line), the record might stay.

**Review by Khalid-Sultan — APPROVED** [review]
